### ISSUE TITLE
[BUG]: HistoricalCryptoBarsRequest WithExchanges clears out BarTimeFrame (and no way to set)

### ISSUE THREAD

**Issue by nathan-datusarator** [alpacahq/alpaca-trade-api-csharp#415]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After creating a `HistoricalCryptoBarsRequest`, the `TimeFrame` is set  When an exchange is filtered by (using `WithExchanges`), the Timeframe object is cleared to default (0Min). 

There is no way to set this value as all fields are read-only and can only be set in the constructor.

### Expected Behavior

Retain the TimeFrame passed in the constructor

### Steps To Reproduce

```cs
var barTimeFrame = new BarTimeFrame(5, BarTimeFrameUnit.Minute);
var request = new HistoricalCryptoBarsRequest(
    symbol, barTimeFrame, TimeInterval.GetInclusiveIntervalFromThat(start))
    .WithPageSize((uint)limit)
    .WithPageToken(token);

if (GeneralUtils.TryMapType<Alpaca.Markets.CryptoExchange>(exchangeName, out Alpaca.Markets.CryptoExchange exchange))
    request = request.WithExchanges(exchange);

var result = fClient.ListHistoricalBarsAsync(request).Result;
```
After the Constructor and first Withs, the TimeFrame is properly set (inspected by Debug).  After the WithExchanges call, it is cleared.

### Environment

- SDK Version: 5.2.2
- OS (version, bitness): Windows 10, 64bit
- .NET SDK (version): 4.7.2
- target process .NET version/bitness: Auto (32 I believe)

**Comment by OlegRa** [comment]
@nathan-datusarator This bug was fixed in version [5.2.3](../releases/tag/sdk-5.2.3) available on NuGet.

**Comment by nathan-datusarator** [comment]
Works in 5.2.3
